lib/controllers/rocket.py

```diff
@@ -590,9 +480,156 @@ async def simulate(
+            length=nose.length,
+            kind=nose.kind,
+            base_radius=nose.base_radius,
+            rocket_radius=nose.rocket_radius,
+        )
+        rocketpy_nose.position = nose.position
+        return rocketpy_nose
+
+    @staticmethod
+    def get_rocketpy_finset(
+        trapezoidal_fins: TrapezoidalFins,
+    ) -> RocketPyTrapezoidalFins:
+        """
+        Get a rocketpy finset object.
+
+        Returns:
+            RocketPyTrapezoidalFins
+        """
+        rocketpy_finset = RocketPyTrapezoidalFins(
+            n=trapezoidal_fins.n,
+            root_chord=trapezoidal_fins.root_chord,
+            tip_chord=trapezoidal_fins.tip_chord,
+            span=trapezoidal_fins.span,
+            cant_angle=trapezoidal_fins.cant_angle,
+            rocket_radius=trapezoidal_fins.radius,
+            airfoil=trapezoidal_fins.airfoil,
+        )
+        rocketpy_finset.position = trapezoidal_fins.position
+        return rocketpy_finset
+
+    @staticmethod
+    def get_rocketpy_tail(tail: Tail) -> RocketPyTail:
+        """
+        Get a rocketpy tail object.
+
+        Returns:
+            RocketPyTail
+        """
+        rocketpy_tail = RocketPyTail(
+            top_radius=tail.top_radius,
+            bottom_radius=tail.bottom_radius,
+            length=tail.length,
+            rocket_radius=tail.radius,
+        )
+        rocketpy_tail.position = tail.position
+        return rocketpy_tail
+
+    @staticmethod
+    def get_rocketpy_parachute(
+        parachute: Parachute, p: int
+    ) -> RocketPyParachute:
+        """
+        Get a rocketpy parachute object.
+
+        Returns:
+            RocketPyParachute
+        """
+        rocketpy_parachute = RocketPyParachute(
+            name=parachute[p].name[0],
+            cd_s=parachute[p].cd_s[0],
+            trigger=eval(parachute[p].triggers[0]),
+            sampling_rate=parachute[p].sampling_rate[0],
+            lag=parachute[p].lag[0],
+            noise=parachute[p].noise[0],
+        )
+        return rocketpy_parachute
+
+    @staticmethod
+    def check_parachute_trigger(expression: str) -> bool:
+        """
+        Check if the trigger expression is valid.
+
+        Args:
+            expression: str
+
+        Returns:
+            bool: True if the expression is valid, False otherwise.
+        """
+
+        # Parsing the expression into an AST
+        try:
+            parsed_expression = ast.parse(expression, mode="eval")
+        except SyntaxError:
+            print("Invalid syntax.")
+            return False
+
+        # Constant case (supported after beta v1)
+        if isinstance(parsed_expression.body, ast.Constant):
+            return True
+        # Name case (supported after beta v1)
+        if (
+            isinstance(parsed_expression.body, ast.Name)
+            and parsed_expression.body.id == "apogee"
+        ):
+            global apogee
+            apogee = "apogee"
```

Comment:
is this line really necessary? Seeting global variables in python like this is usually a code smeel

Reply:
Touché.

**TLDR**
Probably not necessary, it is a legacy code that needs to be refactored. Rocketpy interfaces within the API will be my next refactoring goal, this time I was focusing on the DB interfaces.

**Context**
When I looked at it this week I was also intrigued. So I'm not sure if this is a code duplication from the rocketpy original obj handling (I had to duplicate some of the rocketpy lib code to map the pydantic models into rocketpy native classes) or if it was a workaround for some other thing, It needs investigation and a refactoring for sure.